sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/identifiers.scala

```diff
@@ -105,3 +105,15 @@ case class FunctionIdentifier(funcName: String, database: Option[String])
 object FunctionIdentifier {
   def apply(funcName: String): FunctionIdentifier = new FunctionIdentifier(funcName)
 }
+
+/**
+ * Catalog identifier.
+
+ * @param space Namespace in the catalog
+ * @param name Object name
+ */
+case class CatalogIdentifier(space: CatalogIdentifier.Namespace, name: String)
```

Comment:
I always fail to understand the name `CatalogIdentifier`.

It's an identifier of a table in a certain catalog, instead of an identifier of a catalog. Basically it's still a table identifier.

Reply:
Yes, it is similar to table identifier.
How about renaming `CatalogIdentifier` to `ObectIdentifier`?
And redefines `CatalogIdentifier` as:
```
CatalogIdentifier(catalog: Option[String], ident: ObectIdentifier)
```

Reply:
Unfortunately `ObjectIdentifier` is not ideal either. Any suggestion?
Otherwise we may live with `CatalogIdentifier` that identifies something inside a catalog.

Reply:
I think `CatalogIdentifier` is fine, unless someone can think of a better option.

Reply:
Looks like `Identifier` isn't used. If this is in the v2 namespace, it would be fine to simply call the interface `Identifier`.